src/screens/walletConnect/WalletConnect2Context.tsx

```diff
@@ -189,7 +196,28 @@ export const WalletConnect2Provider = ({
         }
         adapter
           .handleCall(method, params)
-          .then(signedMessage => {
+          .then(async signedMessage => {
+            if (method === 'eth_sendTransaction') {
+              try {
```

Comment:
I think that we can create a function for this, and then call it here.
From line 201 to line 219, that way we can reuse it across the app if, in the future, we implement a normal wallet.